### PR TITLE
gh-93607: document `root` attribute of `iterparse`

### PR DESCRIPTION
**Pull request by Prometheus3375**
Issue https://github.com/python/cpython/issues/93607

* Issue: gh-93607

### CODE REVIEW

File: Doc/library/xml.etree.elementtree.rst [hunk 1]
```diff
@@ -628,6 +628,9 @@ Functions
    for applications where blocking reads can't be made.  For fully non-blocking
    parsing, see :class:`XMLPullParser`.
 
+   Once *source* is fully read, the returned :term:`iterator` object is populated with a ``root``
+   attribute which references to the root element of the resulting XML tree.
```
Comment (slateny) [inline Doc/library/xml.etree.elementtree.rst]: I believe this reads a bit better

```suggestion
   attribute which references the root element of the resulting XML tree.
```

**Review by slateny — APPROVED** [review]
I'm no expert on this, but looks right and testing shows the same:
```
>>> f = io.StringIO('<data><to></to></data>')
>>> iter1 = iterparse(f)
>>> iter1.root
>>> [_ for _ in iter1]
[('end', <Element 'to' at 0x000001EF004B75B0>), ('end', <Element 'data' at 0x000001EF004B7560>)]
>>> iter1.root
<Element 'data' at 0x000001EF004B7560>
```

**Comment by Prometheus3375** [comment]
It can be note that `root` attribute is always present, it is just `None` until the entire file is processed. In other words, `iterparse_object.root is not None` is a correct way to check that file is fully read while `hasattr(iterparse_object, 'root')` is not. Thus, I am suggesting to clarify this point and also move new docs a bit higher.

```
Returns an :term:`iterator` providing ``(event, elem)`` pairs;
it is also populated with a ``root`` attribute set to ``None``.
Once *source* is fully read, the ``root`` attribute references the root element of the resulting XML tree.
```

**Comment by Prometheus3375** [comment]
It can be note that `root` attribute is always present, it is just `None` until the entire source is processed. In other words, `iterparse_object.root is not None` is a correct way to check that file is fully read while `hasattr(iterparse_object, 'root')` is not (if someone uses it for such checks). Thus, I am suggesting to clarify this point and also move the whole note about `root` a bit higher, when return object is discussed.

```
   Returns an :term:`iterator` providing ``(event, elem)`` pairs;
   it is also populated with a ``root`` attribute being set to ``None``.
   Once *source* is fully read, this attribute references the root element of the 
   resulting XML tree.
```
or
```
   Returns an :term:`iterator` with a ``root`` attribute providing ``(event, elem)`` pairs.
   ``root`` attribute is set to ``None``, but once *source* is fully read, 
   it references the root element of the resulting XML tree.
```

Both variants have weird wording in my opinion, but I couldn't think of a more acceptable one; suggestions are welcome. Maybe such clarification is not necessary at all.

**Comment by slateny** [comment]
Perhaps 

```
   Returns an :term:`iterator` providing ``(event, elem)`` pairs;
   it has a ``root`` attribute that references the root element of the 
   resulting XML tree once *source* is fully read.
```

I think it may be fine to omit the mention of `None` and make it implicit here, but I wonder if it could also be misleading.

**Comment by Prometheus3375** [comment]
I think your variant is acceptable.

Committed variant 
> Once source is fully read, <...> **object is populated** with a `root` attribute

states (imho) that attribute is added only at the end of parsing. This not correct.

Variant
> it has a ``root`` attribute that references the root element of the resulting XML tree once *source* is fully read

does not hold information when attribute is added, but still notes when this attribute has a valid reference. Therefore, it is fine.

**Review by slateny — APPROVED** [review]

**Pull request merged**